### DISCUSSION TITLE
Sentence for HA Voice.  Assist says it "doesn't understand".

### DISCUSSION THREAD

**Discussion by Doni49** [zachowj/node-red-contrib-home-assistant-websocket#1311] [Q&A]
I use Node Red for my Home Assistant automations.  I added a "Sentence" node to one of my flows.  See the attached screenshot.  The node tells me that it's beta and needs Node Red Integration 2.2.0 or higher.  Unless I'm looking in the wrong places, I do have greater than 2.2.0.  Within HA, I went to Settings>Add-ons>Node Red and it says Version 17.0.7.  Within HA's NodeRed environment, I clicked on the menu and at the bottom, it says Version 3.1.5.  I'm not sure where I'd find a version number other than those two places.

When I try to use the Sentence in HA Assist, it tells me "Sorry I didn't understand that".  I typed it in so there wouldn't be a chance of it just misunderstanding the spoken word.

![NodeRedSentence](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/14239245/b3c32bc6-b071-4d2c-9d9e-4e887cdab38f)

**Comment by zachowj** [comment]
This is the node-red integration <https://github.com/zachowj/hass-node-red>

**Reply by Doni49** [reply]
That's odd.  The dialog box (shown in the screenshot) included a link that brought me here to post questions.

**Reply by zachowj** [reply]
An add-on and integration are two different things. You have the add-on installed, version 17.0.7, but you also need the custom integration installed for some of the Home Assistant nodes to function inside Node-RED.

**Reply by zachowj** [reply]
<https://www.home-assistant.io/docs/glossary/#add-on>

<https://www.home-assistant.io/docs/glossary/#integration>
